### PR TITLE
docs: Remove UI docs

### LINKED ISSUE #6924: docs: Make it clear the UI is not currently prioritized
* https://docs.meltano.com/guide/production#meltano-ui
* https://docs.meltano.com/reference/ui#undefined
* few other places

We should clearly indicate that the UI is deprioritized and likely doesn't work with many of the newer features.

### PR DESCRIPTION
**Pull request by tayloramurphy**
Meltano is focusing on improving the CLI as the primary way for users to control and interact with their Meltano project. Many of the newer features in Meltano (`run`, environments, mappers, etc) and MeltanoHub (api) do not work with the Meltano UI and create a bad user experience. 

In the future we will prioritize having a UI, but it will be rebuilt based on our current and future featureset as a complement to the CLI. We invite discussion from the community on the future UI in [this discussion](https://github.com/meltano/meltano/discussions/6957).

This PR removes the documentation around Meltano UI.

Closes https://github.com/meltano/meltano/issues/6924

### CODE REVIEW

File: docs/src/_reference/command-line-interface.md [hunk 1]
```diff
@@ -1341,53 +1341,6 @@ meltano test <plugin1>:<test-name1> <plugin2>:<test-name2>
 
 The `test` command can accept the `--environment` flag to target a specific [Meltano Environment](https://docs.meltano.com/concepts/environments). The [`default_environment` setting](https://docs.meltano.com/concepts/environments#default-environments) in your `meltano.yml` file will be applied if `--environment` is not provided explicitly.
 
-## `ui`
-
-- `meltano ui`: Start the Meltano UI.
-
```
Comment (aaronsteers) [inline docs/src/_reference/command-line-interface.md:1347]: @tayloramurphy , what do you think of keeping this small reference, but with a deprecation note and a pointer to the troubleshooting section?

**Review by aaronsteers — COMMENTED** [review]
@tayloramurphy - On the whole, I think the vast majority of these changes are great and should definitely be merged. There were places where we were unnecessarily promoting the web UI when the better course for the user is just to use the CLI (or yaml, but that's another topic.) 

I do feel like there are certain parts that I'd like to keep around _somewhere_ as reference - for our internal reference if nothing else.

Also - we know there are some number of users interacting with HTTP endpoints that are the web API. And we (soon!) will start working on a REST API interface which might (or might not) build from this as a foundation.

Perhaps we could redirect all 'UI' references to troubleshooting, while putting any helpful reference info for us or for our users into something like a `_reference/server.md` that is omitted from the left-hand index but linked to from `troubleshooting.md`.

I'm also open to other ideas and suggestions as well. Thanks!

File: docs/src/_concepts/project.md [hunk 2]
```diff
@@ -354,15 +354,6 @@ Meltano's CLI utilizes the following tables:
...
 - `plugin_settings` table: [Plugin configuration](/guide/configuration#configuration-layers) set using [`meltano config <plugin> set`](/reference/command-line-interface#config) or [the UI](/reference/ui) when the project is [deployed as read-only](/reference/settings#project-readonly).
-- `user` table: Users for [Meltano UI](/reference/ui) created using [`meltano user add`](/reference/command-line-interface#user).
-
-The remaining tables in the database are used exclusively by Meltano UI, mostly for authentication and authorization purposes:
-
-- `role`
-- `role_permissions`
-- `oauth`
-- `embed_tokens`
-- `subscriptions`
```
[the remark this answers is not in the document]
Comment (sbalnojan) [reply docs/src/_concepts/project.md:365]: @aaronsteers Why not put them into an issue for removing the rest of the UI functionality?

File: docs/src/_reference/command-line-interface.md [hunk 3]
```diff
@@ -1341,53 +1341,6 @@ meltano test <plugin1>:<test-name1> <plugin2>:<test-name2>
 
 The `test` command can accept the `--environment` flag to target a specific [Meltano Environment](https://docs.meltano.com/concepts/environments). The [`default_environment` setting](https://docs.meltano.com/concepts/environments#default-environments) in your `meltano.yml` file will be applied if `--environment` is not provided explicitly.
```
Comment (tayloramurphy) [inline docs/src/_reference/command-line-interface.md:1343]:
```suggestion

## `ui` (deprecated)

The Metano UI is now deprecated. For more information see our [troubleshooting page](/guide/troubleshooting#meltano-ui).
```

File: docs/src/_reference/command-line-interface.md (hunk 1 above)
Comment (tayloramurphy) [reply docs/src/_reference/command-line-interface.md:1347]: I'm cool with that!

File: docs/src/_concepts/project.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (tayloramurphy) [reply docs/src/_concepts/project.md:365]: @aaronsteers my assumption is that having them in version control should be enough. I've made https://github.com/meltano/meltano/issues/6959 to track to the work to remove the code bits and cross-linked the PR.

**Comment by tayloramurphy** [comment]
> And we (soon!) will start working on a REST API interface which might (or might not) build from this as a foundation.

> Perhaps we could redirect all 'UI' references to troubleshooting, while putting any helpful reference info for us or for our users into something like a _reference/server.md that is omitted from the left-hand index but linked to from troubleshooting.md.

@aaronsteers I don't think I can do the redirects at the header level - I think it's only for the full pages. I can test that though. 

Until we start working more on the API my sense is that it's okay to remove the pages as we'll always have them in version control.

File: docs/src/_concepts/project.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (aaronsteers) [reply docs/src/_concepts/project.md:365]: @sbalnojan and @tayloramurphy - I think the crux of this for me is _how long_ until V3 - which is when I thought we'd _remove_ the code and capability.

I don't see that as coming soon personally, with everything else we have on our plates right now. If it is possibly 6 months out, the cost for us and users in the meanwhile is not insignificant. We could hide this behind a feature flag between now and V3 (forcing a bit of extra effort to get the UI to run), but still we have a period of support we need to be accountable for between now and then.

So, my preference would be:

- Yes, we can _remove_ any and all 'how to guides', 'tutorials', etc. These are things that help new users onboard to the features and discover them for the first time.
- Let's preserve all _reference docs_ that we (Meltano team) and existing users of the features will need in order to support currently-onboarded implementations.

**Comment by aaronsteers** [comment]
> > And we (soon!) will start working on a REST API interface which might (or might not) build from this as a foundation.
> 
> > Perhaps we could redirect all 'UI' references to troubleshooting, while putting any helpful reference info for us or for our users into something like a _reference/server.md that is omitted from the left-hand index but linked to from troubleshooting.md.
> 
> @aaronsteers I don't think I can do the redirects at the header level - I think it's only for the full pages. I can test that though.
> 
> Until we start working more on the API my sense is that it's okay to remove the pages as we'll always have them in version control.

As I note in my comment, I don't want to remove 'reference' materials because it just unnecessarily makes things harder for us to manage. I am fine removing everything else, and I'm totally fine with us removing all top-level nav that would point to the reference. (And of course, we'd have a banner at the top of the reference page that says all the below is being deprecated, etc.)

File: docs/src/_guide/troubleshooting.md [hunk 4]
```diff
@@ -2,6 +2,9 @@
 title: Troubleshooting
 description: Learn what you can do if you need to troubleshoot
 layout: doc
+redirect_from:
+ - /reference/ui
+ - /guide/orchestration#meltano-ui
```
Comment (tayloramurphy) [inline docs/src/_guide/troubleshooting.md:7]:
```suggestion
```
confirmed that you can't redirect from an anchor

**Comment by tayloramurphy** [comment]
@aaronsteers I've updated and moved most of the technical content to https://deploy-preview-6955--meltano.netlify.app/reference/ui Let me know what you think.

File: docs/src/_reference/ui.md [hunk 5]
```diff
@@ -2,11 +2,14 @@
 title: Meltano UI
 description: Learn how to manage and monitor your plugins and pipelines using Meltano UI
```
Comment (aaronsteers) [inline docs/src/_reference/ui.md:3]:
```suggestion
```

```suggestion
title: Meltano UI and REST API Server (Deprecated)
description: Reference page for the deprecated Meltano UI and REST API Server configuration settings.
```

File: docs/src/_reference/ui.md [hunk 6]
```diff
@@ -2,11 +2,14 @@
...
 layout: doc
-redirect_from:
-  - /guide/ui/
-weight: 3
+hidden: true
+toc: false
```
Comment (aaronsteers) [inline docs/src/_reference/ui.md:6]: Okay to enable TOC on this page?

```suggestion
toc: true
```

File: docs/src/_contribute/ui.md [hunk 7]
```diff
@@ -1,126 +0,0 @@
...
-# Develop for the embed app
-export MELTANO_EMBED=1
-
-# Develop for the main app (this is the default)
-export MELTANO_EMBED=0
-
-# Start local development environment
-yarn serve
-</pre>
-</div>
```
Comment (aaronsteers) [inline docs/src/_contribute/ui.md:55]: Difficult to do a thorough comparison but I didn't see this contributing content (`yarn serve`, etc.) and could be valuable to preserve in reference page:

- https://deploy-preview-6955--meltano.netlify.app/reference/ui

**Review by aaronsteers — COMMENTED** [review]
A couple minor changes requested. Thanks for the additional changes here to preserve an internal reference for UI and REST API aspects.

I think there is some content in `docs/src/_contribute/ui.md` which still might be worth preserving but I haven't been able to do a thorough enough review of that page to know how much we would preserve. I do think some of it is important, though, such as how to use `yarn` to serve the repo from the dev project.

Happy to take this on if you'd like. I just wanted to make sure I got back to you back on this PR before you were OOO.

**Review by aaronsteers — CHANGES_REQUESTED** [review]
See above ☝️

**Review by aaronsteers — APPROVED** [review]
@tayloramurphy - Looks good! Thanks for tackling this!

**Pull request merged**